### PR TITLE
OWTF-IG comprehensive update on Enumeration and Discovery

### PR DESCRIPTION
**Pull request by VinhPham2106**
Fixed errors of existing tools, added modern tools for enumeration and discovery
## Description

- OWTF container has network capabilities for nmap (and any other tools that needs them)
- SecList has correct name and location in YAML files
- Added "dirsearch" for Active Content Discovery, stop options on 429 to not flood the host and stall the machine
- Added "amass" for additional source of subdomain enumeration 
- Added "httpx" for quick probing
- Added "gau" for passive URL discovery
- Redo robots.txt analysis as all links used are deprecated (in progress)
## Motivation and Context

Enumeration is more than 50% of the work, so a big focus is put on running dirty and quick tools automatically before digging into further details
## Reviewers

@viyatb @7a 
## How Has This Been Tested?

Docker container is built and all OWTF-IG plugins are run on a chosen host. Confirming the result logged in output files
## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [x ] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -19,16 +19,20 @@ ENV PYTHONDONTWRITEBYTECODE 1
...
     libxslt1-dev libldap2-dev libsasl2-dev libffi-dev net-tools lsof locales-all \
     make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev \
     lbd theharvester tlssled nikto dnsrecon nmap whatweb skipfish dirbuster metasploit-framework wpscan wapiti hydra metagoofil o-saft amass && \
-    apt-get clean && \
+    apt -y install httpx-toolkit nuclei subfinder && \
+    apt clean && \
     rm -rf /var/lib/apt/lists/*
 
+# Nmap needs capabilities to run on docker
+RUN setcap cap_net_admin,cap_net_raw=eip $(which nmap)
```
Comment (viyatb) [inline docker/Dockerfile:34]: did you consider the security implications of adding these capabilities? It makes it easier to do privesc :)

File: owtf/data/conf/general.yaml [hunk 2]
```diff
@@ -320,4 +331,4 @@ PROXYCHAINS:
   - config: TOOL_PROXYCHAINS
     value: /usr/bin/proxychains
   - config: PROXYCHAINS_COMMAND
-    value: echo "@@@PROXYCHAINS_CONFIG@@@" > ./proxychains.conf; @@@TOOL_PROXYCHAINS@@@
+    value: echo "@@@PROXYCHAINS_CONFIG@@@" > ./proxychains.conf; @@@TOOL_PROXYCHAINS@@@
```
Comment (viyatb) [inline owtf/data/conf/general.yaml]: newline missing

**Review by viyatb — CHANGES_REQUESTED** [review]
mostly everything looks good - I'd add a couple comments on how those capabilites change the security assumptions when running or deploying OWTF :)

**Comment by VinhPham2106** [comment]
@viyatb I've addressed the security issue by providing user with options, highlighting potential unsafetyness. Please have a look!

**Comment by viyatb** [comment]
looks good! can you see why the tests are failing?

**Comment by VinhPham2106** [comment]
@viyatb fixed it. It was a minor conflict in requirements files!

**Review by viyatb — APPROVED** [review]

**Comment by viyatb** [comment]
thanks @VinhPham2106!

**Pull request merged**